src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtil.java

```diff
@@ -408,4 +408,44 @@ else if (sibling.getType() == TokenTypes.SINGLE_LINE_COMMENT) {
                 || BlockCommentPosition.isOnPackage(blockComment));
     }
 
+    /**
+     * Gets the content of inline custom tag.
+     *
+     * @param javadocInlineTag javadoc inline tag node.
+     * @return String consisting of the content of inline custom tag.
+     */
+    public static String getContentOfInlineCustomTag(DetailNode javadocInlineTag) {
+        final DetailNode[] childrenOfJavadocInlineTag = javadocInlineTag.getChildren();
+        DetailNode nodeToBeProcessed = childrenOfJavadocInlineTag[2];
+        final DetailNode endingNode =
+                childrenOfJavadocInlineTag[childrenOfJavadocInlineTag.length - 1];
+        final StringBuilder customTagContent = new StringBuilder(520);
+
+        while (nodeToBeProcessed != endingNode) {
+            extractContentInlineCustomTag(nodeToBeProcessed, customTagContent);
+            nodeToBeProcessed = getNextSibling(nodeToBeProcessed);
+        }
+        return customTagContent.toString();
+    }
+
+    /**
+     * Extracts the content of inline custom tag recursively.
+     *
+     * @param node DetailNode
+     * @param customTagContent content of custom tag
+     */
+    private static void extractContentInlineCustomTag(DetailNode node,
+                                                      StringBuilder customTagContent) {
+        final DetailNode[] children = node.getChildren();
+        if (children.length == 0) {
```

Comment:
so, it is either node text, or its children texts. Is there some case where there is a text for node and it has children?

Reply:
Not that I could find one, mostly all nodes that have children do not contain any text.
@nmancus1 Is there any node that follows this kind of pattern- (Some text other than node's name)

```
DESCRIPTION -> Some text other than node's name
 |--TEXT -> link to  
 |--JAVADOC_INLINE_TAG -> Some text other than node's name
 |   |--JAVADOC_INLINE_TAG_START -> { 
 |   |--LINK_LITERAL -> @link 
 |   |--WS -> 
 |   |--REFERENCE -> Some text other than node's name
 |   |   |--PACKAGE_CLASS -> java.lang.Terminator 
 |   |   |--HASH -> # 
 |   |   |--MEMBER -> equals 
 |   |   `--PARAMETERS -> Some text other than node's name
 |   |       |--LEFT_BRACE -> ( 
 |   |       |--ARGUMENT -> java.lang.Object 
 |   |       `--RIGHT_BRACE -> ) 
 |   `--JAVADOC_INLINE_TAG_END -> } 
 `--TEXT ->    
```

Reply:
> Is there any node that follows this kind of pattern...

Relevant grammar from `JavadocParser.g4`:

```antlr

description: (
                  ({!isNextJavadocTag()}? LEADING_ASTERISK)
                  | htmlComment
                  | NEWLINE
                  | text
                  | javadocInlineTag
                  | htmlElement
            )+;

```

It appears that this is possible; whether or not it can happen in practice is another story. The best way to find out would be to write such a Javadoc and parse it.

Reply:
@strkkk @nmancus1 I don't think it will be possible as everything parses as text inside of inline tag eg-
```
/**
 * {@sumaary
 * <!-- This is a HTML comment -->
 *  <ul>
 *   <li>Coffee</li>
 * </ul>
 * **** some stars
 * }
 * <!-- HTML comment outside the tag -->
 *  <ul>
 *   <li>Coffee</li>
 * </ul>
 */
```
Will parse as-
```
JAVADOC -> JAVADOC [0:0]
|--TEXT -> /** [0:0]
|--NEWLINE -> \n [0:3]
|--LEADING_ASTERISK ->  * [1:0]
|--TEXT ->   [1:2]
|--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG [1:3]
|   |--JAVADOC_INLINE_TAG_START -> { [1:3]
|   |--CUSTOM_NAME -> @sumaary [1:4]
|   |--NEWLINE -> \n [1:12]
|   |--LEADING_ASTERISK ->  * [2:0]
|   |--WS ->   [2:2]
|   |--DESCRIPTION -> DESCRIPTION [2:3]
|   |   |--TEXT -> <!-- This is a HTML comment --> [2:3]
|   |   |--NEWLINE -> \n [2:34]
|   |   |--LEADING_ASTERISK ->  * [3:0]
|   |   |--TEXT ->   <ul> [3:2]
|   |   |--NEWLINE -> \n [3:8]
|   |   |--LEADING_ASTERISK ->  * [4:0]
|   |   |--TEXT ->    <li>Coffee</li> [4:2]
|   |   |--NEWLINE -> \n [4:20]
|   |   |--LEADING_ASTERISK ->  * [5:0]
|   |   |--TEXT ->  </ul> [5:2]
|   |   |--NEWLINE -> \n [5:8]
|   |   |--LEADING_ASTERISK ->  * [6:0]
|   |   |--TEXT ->  **** some stars [6:2]
|   |   |--NEWLINE -> \n [6:18]
|   |   |--LEADING_ASTERISK ->  * [7:0]
|   |   `--TEXT ->   [7:2]
|   `--JAVADOC_INLINE_TAG_END -> } [7:3]
|--NEWLINE -> \n [7:4]
|--LEADING_ASTERISK ->  * [8:0]
|--TEXT ->   [8:2]
|--HTML_COMMENT -> HTML_COMMENT [8:3]
|   |--HTML_COMMENT_START -> <!-- [8:3]
|   |--TEXT ->  HTML comment outside the tag  [8:7]
|   `--HTML_COMMENT_END -> --> [8:37]
|--NEWLINE -> \n [8:40]
|--LEADING_ASTERISK ->  * [9:0]
|--TEXT ->    [9:2]
|--HTML_ELEMENT -> HTML_ELEMENT [9:4]
|   `--HTML_TAG -> HTML_TAG [9:4]
|       |--HTML_ELEMENT_START -> HTML_ELEMENT_START [9:4]
|       |   |--START -> < [9:4]
|       |   |--HTML_TAG_NAME -> ul [9:5]
|       |   `--END -> > [9:7]
|       |--NEWLINE -> \n [9:8]
|       |--LEADING_ASTERISK ->  * [10:0]
|       |--TEXT ->     [10:2]
|       |--HTML_ELEMENT -> HTML_ELEMENT [10:5]
|       |   `--LI -> LI [10:5]
|       |       |--LI_TAG_START -> LI_TAG_START [10:5]
|       |       |   |--START -> < [10:5]
|       |       |   |--LI_HTML_TAG_NAME -> li [10:6]
|       |       |   `--END -> > [10:8]
|       |       |--TEXT -> Coffee [10:9]
|       |       `--LI_TAG_END -> LI_TAG_END [10:15]
|       |           |--START -> < [10:15]
|       |           |--SLASH -> / [10:16]
|       |           |--LI_HTML_TAG_NAME -> li [10:17]
|       |           `--END -> > [10:19]
|       |--NEWLINE -> \n [10:20]
|       |--LEADING_ASTERISK ->  * [11:0]
|       |--TEXT ->   [11:2]
|       `--HTML_ELEMENT_END -> HTML_ELEMENT_END [11:3]
|           |--START -> < [11:3]
|           |--SLASH -> / [11:4]
|           |--HTML_TAG_NAME -> ul [11:5]
|           `--END -> > [11:7]
|--NEWLINE -> \n [11:8]
|--LEADING_ASTERISK ->  * [12:0]
|--TEXT -> / [12:2]
`--EOF -> <EOF> [12:3]
```
They are parsed as HTML tags outside the inline tag, inside they are just parsed as text, also outside the inline tag the parent nodes do not have any text which is not contained by the nested/children nodes.

Reply:
ok, let it be like this, in case of some case we didnt expect, fix will be easy